### DISCUSSION TITLE
[New UI] Add CLI option to disable new UI per-command

### DISCUSSION THREAD

**Discussion by mogzol** [vercel/turborepo#8361] [Ideas]
### Goals

1. Allow the new UI to be disabled per-command via a CLI option and/or an environment variable

### Background

The new UI is great, but there are some cases where I would prefer to have the old, non-interactive UI. There is an option to disable it globally (https://turbo.build/repo/docs/reference/configuration#ui), but I don't want to disable it everywhere, just in certain scenarios.

Currently you can kind of work around this by piping the output into `cat`, which forces the old ui, but you lose colors unless you also set the `FORCE_COLOR` env variable:

```
FORCE_COLOR=1 turbo run build | cat
```

This seems like a hacky workaround for what should be a configurable option.

### Proposal

In addition to the global `ui` option, allow it to be configured per-command via a CLI option like:

```
turbo run build --ui=stream
```

And/or via an environment variable like:

```
TURBO_UI=stream turbo run build
```

**Comment by mogzol** [comment]
As of v2.0.2, it looks like the `log-order` CLI option achieves this, thanks to #8343. Note that it must come **BEFORE** `run` if you are using the `turbo run` command (which seems like a bug, I made a comment about it in that PR):

```
turbo --log-order=stream run build
```

Alternatively, if you just use `turbo` without `run` then it works in any position:

```
turbo build --log-order=stream
```

Either way, I'm going to close this since there is a solution now.

**Reply by igloo1505** [reply]
My lord. I love Vercel and use turborepo and Next almost daily, but that CLI decision must've come at 3am. That was a nightmare and made the log output pretty much useless.

**Reply by tclass** [reply]
yep, have the same problem and would really like an env var to turn this off

**Comment by 0x33dm** [comment]
IMHO the new UI should never been the default in the first place.

i tried to CTRL+C i tried so many things and i feel like "it just takes over".

**Reply by LegendKaito-Q** [reply]
I agree 100%. I'm trying to copy a JSON output from the logs and it doesn't allow me to by clicking and dragging the mouse.

**Reply by 0x33dm** [reply]
@itechKaitoQueiroz did you try `turbo --log-order=stream` this seem to revert back to previous working version.
